src/Community/Community.jl

```diff
@@ -1,6 +1,7 @@
 module Community
 
 using LightGraphs
+using LightGraphs.SimpleGraphsCore: SimpleGraph
```

Comment:
The current state of LG is that SimpleGraphs is completely integrated into the library. This causes a bunch of problems due to assumptions of availability. This PR originally started off as a way to isolate SimpleGraphs from LightGraphs proper. This turned out to be problematic, because there are some library functions that need the ability to create some sort of graph structure internally in order to do calculations. (An example is `Community/clique_percolation`). This causes problems because we now have a dependency loop: SimpleGraphs includes generators, which depend on Community, which depends on SimpleGraphs.

The way around this mess is to create `SimpleGraphsCore`, which is a library that defines SimpleGraphs, SimpleDiGraphs, and SimpleEdges, constructors, the approved interface, and some core functions. This module is NOT intended for users - it is used only by the LightGraphs library functions that need to create some graphs. It does not have any of the generators, or anything fancy. It also doesn't export anything.

`SimpleGraphs` is now its own module, and what it does is import `SimpleGraphsCore`, add some generators and operator overrides, and then reexport the whole thing for users.

Also gone are the generically named `Graph`, `DiGraph`, and `Edge`. These are deprecated in favor of their `Simple*` equivalents.